server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -1569,6 +1569,11 @@ private Node stop() {
     // In this case the process will be terminated even if the first call to close() has not finished yet.
     @Override
     public synchronized void close() throws IOException {
+        logger.info("Invoking termination handler plugins...");
+        pluginsService.filterPlugins(NodeTerminationHandlerPlugin.class).forEach(plugin -> {
```

Comment:
Don't we need this to happen after the readiness service is stopped, so that the proxy stops sending requests while we shutdown the node?

Reply:
Probably! I only have a vague understanding of the readiness service/proxy interaction. I'll look into this, in the meantime do you know of any docs/presentations that outline this?